plugins/opensslcryptoplugin/cryptoplugin_common.cpp

```diff
@@ -101,6 +101,26 @@ namespace {
         }
         return 0; // no cipher sessions available.
     }
+
+    void resizeInitializationVector(QByteArray *initVector,
+                                    Sailfish::Crypto::CryptoManager::BlockMode blockMode,
+                                    Sailfish::Crypto::CryptoManager::Algorithm algorithm)
+    {
+        if (!initVector || blockMode == Sailfish::Crypto::CryptoManager::BlockModeEcb) {
+            // IV not required for this mode
+            return;
+        }
+
+        // Ensure the IV has the correct size. The IV size for most modes is the same as the block size.
+
+        switch (algorithm) {
+        case Sailfish::Crypto::CryptoManager::AlgorithmAes:
+            initVector->resize(16);  // AES = 128-bit block size
```

Comment:
In the case where the initialisation vector is too small, we should fill it with randomly generated bytes and emit the generated IV as per https://github.com/sailfishos/sailfish-secrets/blob/master/lib/Crypto/cipherrequest.cpp#L463.

In the case where the initialisation vector is too big, I guess we can truncate, but we still need to notify the client that the IV was "generated" / changed, so that they can send the "correct" IV to the recipient etc.